libs/sdk-react-native/ios/BreezSDKMapper.swift

```diff
@@ -195,6 +195,47 @@ class BreezSDKMapper {
         }
     }
     
+    static func asOpeningFeeParams(reqData: [String: Any]) -> OpeningFeeParams? {
+        if let minMsat = reqData["minMsat"] as? UInt64,
+           let proportional = reqData["proportional"] as? UInt32,
+           let validUntil = reqData["validUntil"] as? String,
+           let maxIdleTime = reqData["maxIdleTime"] as? UInt32,
+           let maxClientToSelfDelay = reqData["maxClientToSelfDelay"] as? UInt32,
+           let promise = reqData["promise"] as? String,{
+            return OpeningFeeParams(minMsat: minMsat, proportional: proportional, validUntil: validUntil, maxIdleTime: maxIdleTime, maxClientToSelfDelay: maxClientToSelfDelay, promise: promise)
+        }
+        
+        return nil
+    }
+    
+    static func asReceivePaymentRequest(reqData: [String: Any?]) -> ReceivePaymentRequest? {
+        if let amountSats = reqData["amountSats"] as? UInt64,
+           let description = reqData["description"] as? String
+        {
+            let preimage = reqData["preimage"] as? [UInt8]
+            let openingFeeParams = asOpeningFeeParams(reqData: reqData["openingFeeParams"])
+            return ReceivePaymentRequest(amountSats: amountSats, description: description, preimage: preimage, openingFeeParams: openingFeeParams)
+        }
+        
+        return nil
+    }
+    
+    static func asReceiveOnchainRequest(reqData: [String: Any?]) -> ReceiveOnchainRequest {
+        let openingFeeParams = asOpeningFeeParams(reqData: reqData["openingFeeParams"])
+        return ReceiveOnchainRequest(openingFeeParams: openingFeeParams)
+        
+    }
+    
+    static func asBuyBitcoinRequest(reqData: [String: Any?]) -> BuyBitcoinRequest? {
+        if let buyBitcoinProvider = asBitcoinProvider(reqData"buyBitcoinProvider"),
+        {
+            let openingFeeParams = asOpeningFeeParams(reqData: reqData["openingFeeParams"])
+            return BuyBitcoinRequest(buyBitcoinProvider: buyBitcoinProvider, openingFeeParams: openingFeeParams)
+        }
+        
+        return nil
```

Comment:
First the provider needs typing to a string and init BuyBitcoinRequest with `provider: buyBitcoinProvider`
```
        if let provider = reqData["provider"] as? String {
            let buyBitcoinProvider = asBitcoinProvider(provider: provider)
            let openingFeeParams = asOpeningFeeParams(reqData: reqData["openingFeeParams"])
            return BuyBitcoinRequest(provider: buyBitcoinProvider, openingFeeParams: openingFeeParams)
        }

        return nil

Reply:
Done in 659b6a5abe4a04154060b995fe81dbb21c2c05c3